lib/app/dispatcher.rb

```diff
@@ -0,0 +1,92 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+# frozen_string_literal: true
+
+require 'active_support'
+require 'concurrent-ruby'
+require 'connectors'
+require 'core'
+require 'utility'
+require 'app/config'
+
+module App
+  class Dispatcher
+    POLL_IDLING = (App::Config[:idle_timeout] || 60).to_i
+    TERMINATION_TIMEOUT = (App::Config[:termination_timeout] || 60).to_i
+
+    attr_reader :workers
+
+    def initialize
+      @workers = {}.with_indifferent_access
+      @pool = Concurrent::ThreadPoolExecutor.new(
+        min_threads: 5,
+        max_threads: 10,
+        max_queue: 100,
+        fallback_policy: :abort
+      )
+      @is_shutting_down = false
+    end
+
+    def start!
+      loop do
+        connectors = Core::ElasticConnectorActions.native_connectors
+        if connectors.empty?
+          Utility::Logger.info('No native connectors found.')
+          return
```

Comment:
Should not return, but go to next loop, after sleeping.

Reply:
Right. Fixing.